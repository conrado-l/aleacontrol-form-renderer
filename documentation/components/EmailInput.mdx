---
name: EmailInput
---

import { Playground, PropsTable } from 'docz'
import EmailInput from '../../src/components/EmailInput.js'
import 'bootstrap/dist/css/bootstrap.min.css'

# EmailInput

<PropsTable of={EmailInput} />

## Description
A simple email input
## Demo and usage

<Playground>
    <EmailInput value='info@aleacontrol.com' name='email' col={4} update={(e) => e}/>
</Playground>
