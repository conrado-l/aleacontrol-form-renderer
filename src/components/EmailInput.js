import React, { Component } from 'react'
import PropTypes from 'prop-types'

class EmailInput extends Component {
  render () {
    return (
      <input type='email'
        className={`form-control col-xs-12 col-md-${this.props.col}`}
        name={this.props.name}
        value={this.props.value}
        onChange={(e) => this.props.update(this.props.name, e.target.value)} />
    )
  }
}

EmailInput.propTypes = {
  name: PropTypes.string,
  value: PropTypes.string,
  col: PropTypes.number,
  update: PropTypes.func
}
export default EmailInput
