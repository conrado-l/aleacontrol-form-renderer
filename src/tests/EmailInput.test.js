import React from 'react'
import { mount } from 'enzyme'
import EmailInput from '../components/EmailInput'
import renderer from 'react-test-renderer'

const inputFn = jest.fn()
describe('EmailInput', () => {
  const mockupProps = {
    name: 'email',
    value: 'paul@gmail.com'
  }

  it('should render correctly email component', () => {
    const EmailInputComponent = renderer
      .create(<EmailInput name={mockupProps.name} value={mockupProps.value} />)
      .toJSON()
    expect(EmailInputComponent).toMatchSnapshot()
  })

  it('should render the input with the proper value and name', () => {
    const component = mount(<EmailInput name={mockupProps.name} value={mockupProps.value} />)

    expect(
      component.containsMatchingElement(
        <input type='email' value={mockupProps.value} name={mockupProps.name} />
      )
    ).toBeTruthy()
  })

  it('should call the update callback function with proper parameters on input', () => {
    const EmailInputComponent = mount(<EmailInput name={mockupProps.name} value={mockupProps.value} update={inputFn} />)
    const inputString = 'Aleacontrol'

    EmailInputComponent.find('input').simulate('change', {
      target: { value: inputString }
    })

    expect(inputFn).toHaveBeenCalledWith(mockupProps.name, inputString)
  })
})
