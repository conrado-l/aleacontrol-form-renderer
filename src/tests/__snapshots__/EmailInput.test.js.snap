// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailInput should render correctly email component 1`] = `
<input
  className="form-control col-12 col-sm-undefined "
  onChange={[Function]}
  type="email"
/>
`;
